### PR TITLE
deploymentLabels for stateful set

### PR DESCRIPTION
**Pull request by respondsid**
deploymentLabels for stateful set

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@respondsid Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-deployer-kubernetes&pullRequestId=379)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-cloud/spring-cloud-deployer-kubernetes&pullRequestId=379) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@respondsid Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-cloud/spring-cloud-deployer-kubernetes&pullRequestId=379)!

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java [hunk 1]
```diff
@@ -356,8 +356,8 @@ protected void createStatefulSet(String appId, AppDeploymentRequest request, Map
 				.addToMatchLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE).endSelector()
 				.withVolumeClaimTemplates(persistentVolumeClaimBuilder.build()).withServiceName(appId)
 				.withPodManagementPolicy("Parallel").withReplicas(replicas).withNewTemplate().withNewMetadata()
-				.withLabels(idMap).addToLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE).endMetadata().withSpec(podSpec)
-				.endTemplate().build();
+				.withLabels(idMap).addToLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE).addToLabels(getDeploymentLabels(request))
+				.endMetadata().withSpec(podSpec).endTemplate().build();
 
 		StatefulSet statefulSet = new StatefulSetBuilder().withNewMetadata().withName(appId).withLabels(idMap)
 				.addToLabels(SPRING_MARKER_KEY, SPRING_MARKER_VALUE).endMetadata().withSpec(spec).build();
```
Comment (ilayaperumalg) [inline src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java:363]: I think we also need to set the `label` on StatefulSet `metadata` here.

**Comment by ilayaperumalg** [comment]
@respondsid Thanks for the PR!

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java (hunk 1 above)
Comment (respondsid) [reply src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java:363]: Ok let me see, I did verify all the labels got transferred over to the individual plods. I can check again

Comment (respondsid) [reply src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesAppDeployer.java]: @ilayaperumalg  thanks for review, I added the labels in StatefulSet metadata, I also added verification for all Stateful set metadata, pod template and pod metadata. Please review

**Comment by ilayaperumalg** [comment]
Hi @respondsid, Thanks for the update! The changes look good and merging.

**Comment by respondsid** [comment]
@ilayaperumalg thanks for merging. Quick question how can i pull spring cloud dataflow with this fix into my env ?

**Comment by ilayaperumalg** [comment]
Both the Skipper and SCDF are using the latest SNAPSHOT from the deployer-kubernetes (2.3.0.BUILD-SNAPSHOT). If you have installed the `spring-cloud-deployer-kubernetes` in your local via `./mvnw install`, both the SCDF and Skipper should be able to pull the latest.

**Comment by respondsid** [comment]
@ilayaperumalg  i am using kubernetes deployment files to pulling images directly from docker hub. I see a snapshot build for docker pull springcloud/spring-cloud-dataflow-server:2.5.0.BUILD-SNAPSHOT my understanding is this include spring cloud deployer 1.3.0 snapshot release ? So I can use this docker image to deploy on my setup ?

**Pull request merged**